frontend/src/scene/payment/state.js

```diff
@@ -0,0 +1,35 @@
+import { observable, decorate, action } from 'mobx';
+import { getPaymentDetails } from '../../apis';
+
+class PaymentFormState {
+    order_number = '';
+    submitError = true;
+    submitSuccess = false;
+
+    constructor(userStore) {
+        this.userStore = userStore;
+        this.balance = this.userStore.balance;
+    }
+
+    paymentDetails = async (orderNumber) => {
```

Comment:
all the information has been passed from the server. It's redundant to ask the server again.
In case of network failure, user will wonder why they paid, get redirected, and nothing show up

Reply:
fixed